internal/api/sms_provider/huawei_cloud.go

```diff
@@ -0,0 +1,259 @@
+package sms_provider
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+
+	"fmt"
+	"io"
+
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/supabase/gotrue/internal/conf"
+)
```

Comment:
please trim the empty lines in the import segment, and re-format them with `go fmt`